ProcessLib/HT/StaggeredHTFEM-impl.h

```diff
@@ -0,0 +1,286 @@
+          unsigned GlobalDim>
+void StaggeredHTFEM<ShapeFunction, IntegrationMethod, GlobalDim>::
+    assembleHeatTransportEquation(double const t,
+                                  std::vector<double>& local_M_data,
+                                  std::vector<double>& local_K_data,
+                                  std::vector<double>& /*local_b_data*/,
+                                  LocalCoupledSolutions const& coupled_xs)
+{
+    auto const& local_p = coupled_xs.local_coupled_xs[1];
+    auto const local_matrix_size = local_p.size();
+    // This assertion is valid only if all nodal d.o.f. use the same shape
+    // matrices.
+    assert(local_matrix_size == ShapeFunction::NPOINTS);
+
+    auto local_p_Eigen_type =
+        Eigen::Map<const NodalVectorType>(&local_p[0], local_matrix_size);
+
+    auto const& local_T1 = coupled_xs.local_coupled_xs[0];
+
+    auto local_M = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+
+    SpatialPosition pos;
+    pos.setElementID(this->_element.getID());
+
+    auto const& material_properties = this->_material_properties;
+
+    auto const& b = material_properties.specific_body_force;
+
+    GlobalDimMatrixType const& I(
+        GlobalDimMatrixType::Identity(GlobalDim, GlobalDim));
+
+    MaterialLib::Fluid::FluidProperty::ArrayType vars;
+
+    unsigned const n_integration_points =
+        this->_integration_method.getNumberOfPoints();
+
+    for (std::size_t ip(0); ip < n_integration_points; ip++)
+    {
+        pos.setIntegrationPoint(ip);
+
+        auto const& ip_data = this->_ip_data[ip];
+        auto const& N = ip_data.N;
+        auto const& dNdx = ip_data.dNdx;
+        auto const& w = ip_data.integration_weight;
+
+        double p_at_xi = 0.;
+        NumLib::shapeFunctionInterpolate(local_p, N, p_at_xi);
+        double T1_at_xi = 0.;
+        NumLib::shapeFunctionInterpolate(local_T1, N, T1_at_xi);
+
+        auto const porosity =
+            material_properties.porous_media_properties.getPorosity(t, pos)
+                .getValue(t, pos, 0.0, T1_at_xi);
+
+        vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::T)] =
+            T1_at_xi;
+        vars[static_cast<int>(MaterialLib::Fluid::PropertyVariableType::p)] =
+            p_at_xi;
+        auto const fluid_density =
+            material_properties.fluid_properties->getValue(
+                MaterialLib::Fluid::FluidPropertyType::Density, vars);
+        auto const specific_heat_capacity_fluid =
+            material_properties.fluid_properties->getValue(
+                MaterialLib::Fluid::FluidPropertyType::HeatCapacity, vars);
+
+        // Assemble mass matrix
+        local_M.noalias() +=
+            w *
+            this->getHeatEnergyCoefficient(t, pos, porosity, fluid_density,
```

Comment:
Again, I'm not sure if `this` is required here...

Reply:
It is needed in the this derived class.